pkg/joinserver/redis/registry.go

```diff
@@ -92,11 +110,21 @@ func (r *DeviceRegistry) SetByEUI(ctx context.Context, joinEUI types.EUI64, devE
 		if err != nil {
 			return err
 		}
+		if stored == nil && pb == nil {
+			return nil
```

Comment:
Shouldn't this be a not found?
Also, this behavior would need to be replicated in AS and NS registries

Reply:
This is a no-op; you don't set an entity that doesn't exist.

Replicate across other registries is fine, I'll do that.

Reply:
Can you explain why it's not a "not found" error?
You're deleting an entity, which does not exist - that sounds like a 404 to me. @htdvisser ?

Reply:
If you want a not found error, you can do that in the callback, right? There you have the context of whether you expect something to exist or not. Isn't this how it's implemented everywhere?